Dockerfile

```diff
@@ -0,0 +1,16 @@
+# Copyright The OpenTelemetry Authors
```

Comment:
This seems to fit better in the `example/http` directory as it runs that example.

Reply:
Usually it is easier to keep the `Dockerfile` and `docker-compose.yml` here at the top of the directory structure so the build context and all files required to build the examples are specified as subdirectories relative to current directory. It makes the invocation of docker easiest and is the most typical:
```
$ docker build . # the typical way users expect to build docker images
```
If the `Dockerfile` were moved into the `example/http` directory, the location of the `Dockerfile` would have to be specified explicitly instead of using default location to keep the build context pointed at the current directory.
```
$ docker build -f example/http/Dockerfile . # use current dir as build context, but have to point to non-standard location for Dockerfile
```
If the user were to change into the `example/http` directory to use `Dockerfile` in there, would have to then know to specify grandparent directory as build context so the layers could find the files required to build the example binary.
```
$ cd example/http
$ docker build ../.. # use grandparent directory as build context, which is not usual but works
```
In other words, I recommend leaving the `Dockerfile` and `docker-compose.yml` at top of directory structure, but if you prefer it moved into the subdirectory, I can do that and add extra documentation to explain proper invocation.

Reply:
I'd second that, for both the `Dockerfile` and `docker-compose.yml`.  It leaves room to add further examples with their own build and run configurations.  I may be the exception, but I've grown used to finding `Dockerfile`s in all sorts of places other than the top-level of a repository.  I think the critical thing is to ensure that it's clearly documented what context the `Dockerfile` expects to receive, whether that's the root of the repository or the `example/http` directory, or something different.

Reply:
How will we add Dockerfiles to run the prometheus, gRPC, namedtracer, or any other added examples?

Reply:
I'll split up the stages of the current `Dockerfile` into just what is required for `example/http` and `example/zipkin` and put into their respective directories and update the docs.

Reply:
@Aneurysm9 @MrAlias the git commits 797ec0e 117883b should resolve this change request